### ISSUE TITLE
tune_sim_anneal not using event_level = "second" in control?

### ISSUE THREAD

**Issue by patelis** [tidymodels/finetune#40]
I'm trying to run tune_sim_anneal to search for better parameters for a xgboost model after having run a normal tune first. 
Snippet of what the code looked like below:

```{r}

folds <- vfold_cv(train, v = 5, strata = target)

rec <- recipe(target ~ ., data = train) %>% 
  update_role(customer_ID, new_role = "id") %>% 
  step_normalize(all_numeric_predictors()) %>% 
  step_dummy(all_nominal_predictors()) %>% 
  step_nzv() #%>% 

xgb_model <- boost_tree(
    trees = 2000, 
    learn_rate = tune(), # 0.0275,  
    min_n = tune(), #22L,
    tree_depth = tune(), #11L,
    mtry = tune(), #616L,
    stop_iter = 20L
  ) %>%
  set_engine('xgboost', event_level = "second", 
                     tree_method = "gpu_hist", gpu_id = 0,
                     nthread = 10) %>%
  set_mode('classification')

xgb_wf <- workflow() %>% 
  add_recipe(rec) %>% 
  add_model(xgb_model)

xgb_params <- xgb_wf %>%
  extract_parameter_set_dials() %>%
  update(mtry = mtry(range(600L, 1200L)),
         tree_depth = tree_depth(range(4L, 8L)),
         learn_rate = learn_rate(range(0.025, 0.035), trans = NULL),
         min_n = min_n(range(15L, 30L))
  )

prob_metrics <- metric_set(mean_coeff) # This is a custom metric I defined as instructed from the yardstick vignettes.
param_grid <- grid_latin_hypercube(xgb_params, size = 10)

xgb_tune <- xgb_wf %>%
  tune_grid(
    folds,
    param_info = xgb_params,
    grid = param_grid,
    metrics = prob_metrics,
    control = control_grid(
      verbose = TRUE,
      save_pred = TRUE,
      save_workflow = TRUE,
      event_level = "second"
    )
  )

xgb_race <- tune_sim_anneal(
  object = xgb_wf,
  resamples = folds,
  iter = 20,
  param_info = xgb_params,
  metrics = prob_metrics,
  initial = xgb_tune,
  control = control_sim_anneal(
    verbose = TRUE,
    save_pred = TRUE,
    save_workflow = TRUE,
    event_level = "second"
  )
)

``` 
The data and metric are coming from: https://www.kaggle.com/competitions/amex-default-prediction/overview/evaluation
The metric provides a different value depending on whether the event_level is first or second (in my case event_level is second).

The problem is that event_level = "second" does not seem to register with tune_sim_anneal(). 
This is the output I get after running collect_metrics():

<img width="469" alt="image" src="https://user-images.githubusercontent.com/40008856/175809569-71bd8dee-91a4-40e0-94cd-b5bb327f827a.png">

While the parameters in the iteration process are not vastly different, the results are significantly and consistently worse.
As a test, I ran 
```{r}
xgb_race %>% collect_predictions() %>% group_by(.config) %>% mean_coeff(target, .pred_0)
```
to test whether the metric was applied incorrectly.
Indeed, for the initial set of tuning combinations (before the iterative process), the metric estimate was:
<img width="212" alt="image" src="https://user-images.githubusercontent.com/40008856/175809737-c1497c74-811f-4edf-9683-9ee415147dd2.png">

which leads me to believe the event_level is ignored in finetune.
At this point I don't think my custom metric has an issue, because I'd observe the same behaviour in tune_grid() and tune_sim_anneal(), but I could be wrong.

One additional point, even though I had the setup above (+ a couple set.seed() that I removed), when running collect_predictions() on xgb_race, I don't get the predictions for any of the iterations. Is this a different issue or is it because none of the solutions outperform the initial set of estimates provided by tune_grid()?

**Comment by juliasilge** [comment]
It looks to me like you are right on here, and that the `control` object that contains the info about `event_level` isn't passed through to the tuning inside of `tune_sim_anneal()`, like here:

https://github.com/tidymodels/finetune/blob/31a780ccd4eac0accb1aac5f95adcf00446b3812/R/tune_sim_anneal.R#L381-L385

By contrast, it _is_ [passed through in `tune_race_anova()`](https://github.com/tidymodels/finetune/blob/31a780ccd4eac0accb1aac5f95adcf00446b3812/R/tune_race_anova.R#L260-L266) as well as `tune_race_win_loss()`; have you tried either of those with your example?

I _think_ you are right on the second point as well (about not getting predictions) but let's see what happens after the first problem is resolved.

**Comment by patelis** [comment]
I tried `tune_race_anova()` and it worked as expected including saving the predictions in the object, it seems you've pinpointed the issue with `tune_sim_anneal()`.
